Makefile

```diff
@@ -0,0 +1,62 @@
+.PHONY: docker docker-itest docker-test docker-test-all docker-check docker-shell itest test test-all
+
+IMG_NAME      := lndsigner-builder
+
+GOVER         := 1.19.5
+GOPLATFORM    := amd64
+CPPPLATFORM   := x86_64
+LND           := v0.15.5-beta
+BITCOIND      := 24.0.1
+VAULT         := 1.12.2
+
+STAMPPREIMAGE := $(GOVER)$(GOPLATFORM)$(CPPPLATFORM)$(LND)$(BITCOIND)$(VAULT)
+
+BUILDERSTAMP  != echo $(STAMPPREIMAGE) | sha256sum | cut -d " " -f 1
+IMAGE_BUILT   != docker image ls $(IMG_NAME):$(BUILDERSTAMP) | grep $(BUILDERSTAMP) | wc -l
+
+# docker-check checks if an image with the builderstamp already exists. If not,
+# it builds one.
+docker-check: 
+ifeq ($(IMAGE_BUILT), 0)
+	docker build -t $(IMG_NAME):$(BUILDERSTAMP) .
+endif
+
+# docker just tags the latest image to the builderstamp, in case the
+# dependencies have been changed and a new image was built.
+docker: docker-check
+	docker image tag $(IMG_NAME):$(BUILDERSTAMP) $(IMG_NAME):latest
+
+# docker-itest runs itests in a docker container, then removes the container.
+docker-itest: docker
+	docker run -t --rm \
+		--mount type=bind,source=$(CURDIR),target=/app $(IMG_NAME):latest \
+		make itest
+
+# docker-test runs unit tests in a docker container, then removes the container.
+docker-test: docker
+	docker run -t --rm \
+		--mount type=bind,source=$(CURDIR),target=/app $(IMG_NAME):latest \
+		make test
+
+# docker-test-all runs unit and integration tests in a docker container, then
+# removes the container.
+docker-test-all: docker
```

Comment:
Looks like OSX is missing `sha256sum` by default. I've reworked the Makefile according to https://stackoverflow.com/questions/42123064/generate-and-check-download-file-checksum-in-makefile - hopefully that will work, but I have no OSX device on which to test.

Reply:
I do have `sha256sum`, and your change also doesn't fix the problem for me on osx.

I tried `$(shell echo $(STAMPPREIMAGE) | $(SHA256_CMD) | cut -d " " -f 1)` which at least lets that variable be populated properly. It does run into a next problem then though.

```
error obtaining VCS status: exit status 128
	Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
	Use -buildvcs=false to disable VCS stamping.
```

Is it actually worth to implement your own caching mechanism instead of relying on docker's default image build cache?

Reply:
I just took the custom caching out after playing around with the default docker image build cache. Thanks for the pointer, I didn't realize it existed!

Reply:
I finally ran into these errors on my system and the CI, and fixed it in the Dockerfile in https://github.com/bottlepay/lndsigner/compare/29e80e9a7433fdf30993f7cec87b5c012b2a753a..c8d4e2145dcf4ce6412f5b53ebd0f5b7422f79f4

Reply:
I've added the `-buildvcs=false` option to `go install` in the `Makefile` and that should get around the error for now.